build/release/bootstrap-env.sh

```diff
@@ -0,0 +1,180 @@
+
+# Import Scripts
+source "$(dirname '$0')/logging.sh"
+
+# Basic information
+SCRIPT_NAME="$(basename ${BASH_SOURCE[0]})"
+debugging "Script Name is ${SCRIPT_NAME}"
+
+###############################################################################
+# Function Declaration
+
+# set_build_variable - take 3 parameters 
+# the three parameters are 
+function set_build_variable {
+    if [ ! -z "${3}" ] # check to see if the variable is already not empty.
+    then 
+        if [ ! -z "${3}" ]
+        then 
+            outputEnvironment "${1}=${3}"
+        else
+            outputEnvironment "${1}=${2}"
+        fi 
+        debugging "Environment Variable is set [${1}]"
+    else 
+        debugging "Environment Variable is already set [${1}]"
+    fi
+    
+}
+
+# set_build_type - sets the type of the build and the build id
+# 1 - tag 
+# 2 - github_ref
+# 3 - github_sha 
+function set_build_type {
+    if [ "${BUILD_ID}" == *"Integration"* ] || [ -z "${BUILD_ID}" ]
+    then
+        if [ "${1}-" != "---" ]
+        then 
+            # not empty and is tag build
+            if [[ "${1}" == *"RC"* ]]
+            then
+                outputEnvironment "export BUILD_TYPE=\"RELEASE_CANDIDATE\""
+                info "Build_Type is RELEASE_CANDIDATE"
+            else
+                outputEnvironment "export BUILD_TYPE=\"RELEASE\"" 
+                info "Build_Type is RELEASE"
+            fi
+            set_build_variable BUILD_ID "" "${1}"
+        fi
+    else 
+        # SNAPSHOT build
+        set_build_variable BUILD_ID "" "Integration_${2}_${3}"
+        outputEnvironment "export BUILD_TYPE=\"SNAPSHOT\"" 
+    fi
+}
+
+function outputEnvironment { 
+    echo "export ${1}" >> bootstrap.env
+}
+
+###############################################################################
+# Setup Maven Options 
+
+# Not all JVMs support - -XX:MaxPermSize=512m (don't set it)
+# -Xmx2G -Xms1G 
+if [ ! -f ~/.mavenrc ]
+then 
+    echo "MAVEN_OPTS='-Djava.awt.headless=true '" > ~/.mavenrc
+fi
+
+# Threads that are allowed to run
+# which are passed in with BUILD_THREAD
+# we don't want to embed this in maven_opts as certain goals are not thread safe.
+#
+# Examples: 
+#   1C (1Core)
+#   4  (4 Threads)
+THREAD_COUNT=
+if [ ! -z "${BUILD_THREAD}" ]
+then 
+    THREAD_COUNT="-T ${BUILD_THREAD}"
+fi
+
+###############################################################################
+# Setup Git - makes sure there is a gitignore which puts the log in this folder. 
+debugging "Current directory is $(pwd)"
+
+GIT_IGNORE='.gitignore'
+if [ ! -f ${GIT_IGNORE} ]
+then 
+    echo 'logs/' > ${GIT_IGNORE} 
+    echo '**/.cache' >> ${GIT_IGNORE} 
+fi
+
+###############################################################################
+# wipe out existing bootstrap.env files 
+if [ -f bootstrap.env ]
+then 
+    rm -f bootstrap.env
+fi
+
+###############################################################################
+# Setup Default Variables
```

Comment:
Is this file only used for the TRAVIS build then?  Not github actions build and release?

Reply:
it's for both.  I set it up to execute on both.

Reply:
Never mind, I went back to look at set_build_variable and now I get it.  It only sets the var if the value passed is valued and non-empty.